components/automate-cli/cmd/chef-automate/cleanup.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net"
+	"strings"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+)
+
+var cleanupFlags = struct {
+	onprem        bool
+}{}
+
+// const (
+// 	FSCRIPT = `systemctl stop chef-automate;
+// rm -rf /var/automate-ha;
+// rm -rf /hab`
+// )
+
+func init() {
+	RootCmd.AddCommand(cleanupCmd)
+	cleanupCmd.PersistentFlags().BoolVar(&cleanupFlags.onprem, "onprem", false, "Cleaning up all the instances related to onprem ")
+
+}
+
+var cleanupCmd = &cobra.Command {
+		Use:  "cleanup",
+	    Short: "cleanup the Automate HA instances",
+	    Long:  "cleaning up the instance of all the Automate HA related Applications.",
+		Annotations: map[string]string{
+			NoCheckVersionAnnotation: NoCheckVersionAnnotation,
+		},
+		RunE: runCleanupCmd,
+	}
+
+func runCleanupCmd(cmd *cobra.Command, args []string) error {
+	infra, err := getAutomateHAInfraDetails()
+	if err != nil {
+		return err
+	}
+	sshUser := infra.Outputs.SSHUser.Value
+	sskKeyFile := infra.Outputs.SSHKeyFile.Value
+	sshPort := infra.Outputs.SSHPort.Value
+	writer.Printf(strings.Join(args, ""))
+	if isA2HARBFileExist() {
+		if cleanupFlags.onprem {
+			frontendIps := append(infra.Outputs.ChefServerPrivateIps.Value, infra.Outputs.AutomatePrivateIps.Value...)
+			backendIps := append(infra.Outputs.PostgresqlPrivateIps.Value, infra.Outputs.OpensearchPrivateIps.Value...)
+			args = append(args, "--onprem")
+			// writer.Printf("IPs: " + strings.Join(frontendIps, "") + args[0])
+			for i := 0; i < len(frontendIps); i++ {
+				executeCleanupOnRemote(sshUser, sshPort, sskKeyFile, frontendIps[i], args[0])
+			}
+			for i := 0; i < len(backendIps); i++ {
+				executeCleanupOnBackendRemote(sshUser, sshPort, sskKeyFile, backendIps[i], args[0])
+			}
+
+		}
+	}
+	return nil
+}
+
+func executeCleanupOnRemote(sshUser string, sshPort string, sshKeyFile string, ip string, remoteType string) {
```

Comment:
you need to write a generic functions....same logic has been implemented in config patch command...ideally we can resuse the same function...implement the function once 
and we can pass the args to that ....
func remoteExectuion(username,key,port,ip,command)